test/addon_product_test.rb

```diff
@@ -31,6 +32,16 @@
     end
 
     context "when according to libzypp a product is renamed" do
+      before do
+        subject.main
+        allow(Y2Packager::Repository).to receive(:all).and_return([repo0])
+      end
+
+      let(:repo0) do
+        instance_double(
+          Y2Packager::Repository, repo_id: 0, url: URI("dvd:///sr0"), product_dir: "/p0"
```

Comment:
I'd use some valid URL here, this might be confusing if somebody would copy&paste it to a real system...

```suggestion
          Y2Packager::Repository, repo_id: 0, url: URI("dvd:///?devices=/dev/sr0"), product_dir: "/p0"
```